src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -1,18 +1,25 @@
+///<reference path = "../../page/page.component.ts"/>
 namespace Core {
   export class GeneralPreferencesService {
-    private static readonly DEFAULT_VERTICAL_NAV_STATE = "show";
+    private static readonly DEFAULT_VERTICAL_NAV_STATE = Page.VerticalNavState.Show;
     constructor(private $window: ng.IWindowService) {
       'ngInject';
     }
-    getDefaultVerticalNavState(): string {
+    getDefaultVerticalNavState(): Page.VerticalNavState {
       if (_.isNull(this.$window.localStorage.getItem('defaultVerticalNavState'))) {
         return GeneralPreferencesService.DEFAULT_VERTICAL_NAV_STATE;
+
       } else {
-        return this.$window.localStorage.getItem('defaultVerticalNavState');
+        const defaultVerticalNavState = this.$window.localStorage.getItem('defaultVerticalNavState');
```

Comment:
you dont need to use 'switch' here.
you can simply convert the string to enum and return.